conman/routes/admin.py

```diff
@@ -8,12 +8,18 @@
 """
 
 from django.contrib import admin
+from django.contrib.admin.options import FORMFIELD_FOR_DBFIELD_DEFAULTS
+from django.contrib.admin.widgets import AdminTextInputWidget
 from polymorphic.admin import (
     PolymorphicChildModelAdmin,
     PolymorphicParentModelAdmin,
 )
 
-from .models import Route
+from .models import Route, URLPathField
+
+
+# Have mercy, I can't see an easier way to do this across the whole admin.
+FORMFIELD_FOR_DBFIELD_DEFAULTS[URLPathField] = {'widget': AdminTextInputWidget}
```

Comment:
You could add:

```python
    formfield_overrides = {
        URLPathField: {'widget': AdminTextInputWidget},
    }
```

to both `RouteChildAdmin` and `RouteParentAdmin`.

I think this is less resilient - if someone wants to set `formfield_overrides` for a different field type, they'd lose this override unless they explicitly kept it.

This is true of the other attributes we set though, so I don't know how big a deal this actually is.

Reply:
Yeah, `formfield_overrides` would work for this class in the admin.

I hadn't considered the idea of someone overriding it, but that's a valid concern too. The case I was thinking of is if someone were to define an inline, or similar, it'd go back to the multi-line field, and newline chars would find their way into the field again.